manim/utils/units.py

```diff
@@ -12,26 +14,37 @@ def _size_from_dimension(dim=0, pixels=False):
         return config["frame_height"] if not pixels else config["pixel_height"]
     else:
         # TODO: determine size in z-direction
-        raise ValueError("Dimension not supported.")
+        logging.getLogger("manim").warning(
+            f"Z-direction is not supported for units - returning size of x-dimension."
+        )
+        return config["frame_width"] if not pixels else config["pixel_width"]
 
 
 def _is_dimensional(value):
     return isinstance(value, np.ndarray) and value.shape[0] <= 3
 
 
 class Unit:
-    def __init__(self, value, converter):
+    def __init__(self, value):
         self._value = value
-        self._converter = converter
+        self._converter = lambda val, dim: val
+
+    @property
+    def value(self):
+        return copy.deepcopy(self._value)
+
+    @property
+    def converter(self):
+        return self._converter
```

Comment:
Question: why does converter need to be a property? Shouldn't it be a method?

Reply:
The type of the property is a method (aka `callable(self.converter) == True`). We could access `self._converter` directly but I thought this was a cleaner way of doing so.

Reply:
I understand that it is indeed a callable, but I'm asking why we need to do this
```python
    @property
    def converter(self):
        return self._converter
```
instead of simply this
```python
    def converter(self):
        # gets set at some point during runtime/initialization
        pass
```
and then just do `self.converter = lambda x: x` or something down the line. I mean, I see what you are doing with this dynamical method thing, and it's fine, but I don't see why it needs to be a property.

Reply:
CMIIW, but isn't that usually considered bad practice? The `self.converter = lambda x: x` thing I mean.

Reply:
Well it implicitly is sort of a "if/elif/else" statement. The user creates a specific unit object (par example the `PixelUnit` or `PercentUnit`) which has its own conversion logic. But, it only has to convert one variable type thus avoiding any "if/elif/else" statements to determine what we need to convert to. Instead, `convert` checks what dimension(s) we are dealing with and then calls the `Unit`'s conversion method. 

I think if we turn it into a if/elif/else statement that it goes at the cost of extendability. We would have to extend `convert` each time a unit is added. Now, all we have to do is determine how a certain unit converts and add create a class for it. 

Does that answer your question or did you mean something else?

Also, the lambda expressions could be replaced by an override method. It would matter much but that possibly would be a cleaner way.

Reply:
My question is: does a single Unit instance ever need to change its `convert` method? I think what you are currently doing can be achieved with simple inheritance, no? I'm sorry if I'm missing anything...

Reply:
Nope it doesn't. No idea why I didn't just use a override method. Will do so!

Reply:
I remember why I used the lambda stuff instead of overriding. The `converter` function can this way be defined outside of the class, which would allow other parts of the software to use it as well. Furthermore, I personally don't like the syntax:
```python
def converter(self, value, dim, config):
```
As the function does not use `self`. Now you could argue to remove the `value` parameter and use `self.value`. However, then you'd also have to implement the logic on how to convert an array par example (since then we can determine dimensions automatically) every time you override the function. With the current setup, this only has to be done once in `convert`. 

What are your thoughts on this?

Reply:
I'm not sure I followed entirely, but here are some general thoughts:
1. Right now, it seems like the converter functions are very simple, so much so in fact that I don't foresee them being reused anywhere else in the codebase.
2. This Unit class is more of a special convenience than an actual class. (Especially if we implement the `__mul__` suggestion below. So I'm not really concerned about having `self` be an argument that is never used, because the class itself is awkward in many other ways.
3. If the main functionality doesn't really depend on `self`, then maybe there is a way of implementing this Unit thing as a function and not a class.
4. The point that does concern me a bit is the point you raise about converting arrays each time.

I don't have a straight answer, but these are some of my thoughts. You have more experience with this Unit thing now, what do you think?

This subtle utility classes are always the tricky ones :sweat_smile:

Reply:
@leotrs I implemented various basic math operations for units. Could you give your input on it?